adx/resource_adx_external_table.go

```diff
@@ -0,0 +1,270 @@
+package adx
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/favoretti/terraform-provider-adx/adx/validate"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+type ADXExternalTable struct {
+	Name              string
+	ConnectionStrings string
+	Partitions        string
+	PathFormat        string
+	Folder            string
+	Properties        string
+}
+
+func resourceADXExternalTable() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceADXExternalTableCreateUpdate,
+		UpdateContext: resourceADXExternalTableCreateUpdate,
+		ReadContext:   resourceADXExternalTableRead,
+		DeleteContext: resourceADXExternalTableDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"cluster": getClusterConfigInputSchema(),
+			"database_name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"name": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ForceNew:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"data_format": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
+			},
+
+			"storage_connection_string": {
+				Type:             schema.TypeString,
+				Required:         true,
+				ValidateDiagFunc: validate.StringIsNotEmpty,
```

Comment:
Can we mark this one as sensitive? I assume connection strings can contain secrets?
```suggestion
				ValidateDiagFunc: validate.StringIsNotEmpty,
				Sensitive: true,
```

Reply:
Yes, that would be great. Gonna add this.